starky/src/evaluation_frame.rs

```diff
@@ -0,0 +1,66 @@
+/// A trait for viewing an evaluation frame of a STARK table.
+///
+/// It allows to access the current and next rows at a given step
+/// and can be used to implement constraint evaluation both natively
+/// and recursively.
+pub trait StarkEvaluationFrame<T: Copy + Clone + Default, U: Copy + Clone + Default>:
+    Sized
+{
+    /// The number of columns for the STARK table this evaluation frame views.
+    const COLUMNS: usize;
+    const PUBLIC_INPUTS: usize;
+
+    /// Returns the local values (i.e. current row) for this evaluation frame.
+    fn get_local_values(&self) -> &[T];
+    /// Returns the next values (i.e. next row) for this evaluation frame.
+    fn get_next_values(&self) -> &[T];
+
+    fn get_public_inputs(&self) -> &[U];
+    /// Outputs a new evaluation frame from the provided local and next values.
```

Comment:
```suggestion
    fn get_public_inputs(&self) -> &[U];
    
    /// Outputs a new evaluation frame from the provided local and next values.
```

minor nit, but I'd add a separation here.